receiver/scraperhelper/settings.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package scraperhelper // import "go.opentelemetry.io/collector/receiver/scraperhelper"
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/multierr"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+var (
+	ErrNonPositiveInterval    = errors.New("requires positive value")
+	ErrTimeoutExceedsInterval = errors.New("timeout value exceeds collection interval")
+)
+
+// ScraperControllerSettings defines common settings for a scraper controller
+// configuration. Scraper controller receivers can embed this struct, instead
+// of receiver.Settings, and extend it with more fields if needed.
+type ScraperControllerSettings struct {
+	// CollectionInterval sets the how frequently the scraper
+	// should be called and used as the context timeout
+	// to ensure that scrapers don't exceed the interval.
+	CollectionInterval time.Duration `mapstructure:"collection_interval"`
+	// InitialDelay sets the initial start delay for the scraper,
+	// any non positive value is assumed to be immediately.
+	InitialDelay time.Duration `mapstructure:"initial_delay"`
+	// Timeout is used to set scraper's context deadline, it must be within
+	// the range of (0, CollectionInterval]
+	Timeout time.Duration `mapstructure:"timeout"`
+}
+
+// NewDefaultScraperControllerSettings returns default scraper controller
+// settings with a collection interval of one minute.
+func NewDefaultScraperControllerSettings(component.Type) ScraperControllerSettings {
```

Comment:
The Type seem to not be used, a separate issue to remove it.

Reply:
Yeah, @astencel-sumo were talking about how it is redundant now due to `component.Type` not needing to be embedded into the config. 

What are your thoughts on having available options for setting the default collection interval and initial delay? This would save the awkward initialling of the value to set your component's default collection interval.

Reply:
I don't like that extra public API, if you want to set values, use `ScraperControllerSettings { Timeout: 5s, ...}` no need to use this. We should always avoid extra unnecessary public APIs, because that means extra maintenance for us.

Reply:
> I don't like that extra public API, if you want to set values, use ScraperControllerSettings { Timeout: 5s, ...} no need to use this. 

My first thought was to disagree, but thinking about this more, the `NewDefaultScraperControllerSettings` function defines a default set of settings for the ScraperController that is common to all components. But what is the value of having such a common default set of settings? Shouldn't it be each component defining its own default settings? 🤔 

I wonder if this could be extrapolated to other common functionalities - can we remove exporter helper's [NewDefaultQueueSettings](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.80.0/exporter/exporterhelper/queued_retry.go#L48), [NewDefaultRetrySettings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go#L258), [NewDefaultTimeoutSettings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/common.go#L24)? Can we remove [NewDefaultHTTPClientSettings](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.80.0/config/confighttp/confighttp.go#L82)?